js/base/Exchange.js

```diff
@@ -2654,4 +2654,21 @@ module.exports = class Exchange {
         }
         return undefined;
     }
+
+    handleMarginModeAndParams (methodName, params = {}) {
+        /**
+         * @ignore
+         * @method
+         * @param {object} params extra parameters specific to the exchange api endpoint
+         * @returns {[string|undefined, object]} the marginMode in lowercase as specified by params["marginMode"], this.options["marginMode"] or this.options["defaultMarginMode"]
+         */
+        const defaultMarginMode = this.safeString2 (this.options, 'marginMode', 'defaultMarginMode');
+        const methodOptions = this.safeValue (this.options, methodName, {});
+        const methodMarginMode = (typeof methodOptions === 'string') ? methodOptions : this.safeString2 (methodOptions, 'marginMode', 'defaultMarginMode', defaultMarginMode);
+        const marginMode = this.safeStringLower (params, 'marginMode', methodMarginMode);
```

Comment:
2667th line is incorrect imho. we should no longer support `string` moreover for newly implemented properties. the `string` is only supported for backward-compatibility and it only held value of  **market-type**  when method had a string value in options. so, if a method has string value, we should ignore that totally. So, code would be like this:

```suggestion
        const methodOptions = this.safeValue (this.options, methodName, {});
        const methodMarginMode = (typeof methodOptions === 'string') ? methodOptions : this.safeString2 (methodOptions, 'marginMode', 'defaultMarginMode', defaultMarginMode);
        const marginMode = this.safeStringLower (params, 'marginMode', methodMarginMode);
```